crates/criticaltrust/src/v2/manifests.rs

```diff
@@ -0,0 +1,205 @@
+// SPDX-FileCopyrightText: The Ferrocene Developers
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+//! Serializable and deserializable representation of criticaltrust v2 manifests.
+
+use crate::keys::KeyRole;
+use crate::signatures::Signable;
+use serde::de::Error as _;
+use serde::{Deserialize, Serialize};
+
+/// Typed representation of a manifest version number.
+///
+/// The version number is stored as a const generic rather than as a field of the struct. This is
+/// done to:
+///
+/// * Verify that the version number is correct as part of the deserialization process.
+/// * Simplify constructing manifests: you don't have to specify the version number, type
+///   inference will figure out the right one.
+#[derive(Clone, PartialEq, Eq)]
+pub struct MetadataVersion<const V: u32>;
+
+impl<const V: u32> std::fmt::Debug for MetadataVersion<V> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("MetadataVersion").field(&V).finish()
+    }
+}
+
+impl<const V: u32> Serialize for MetadataVersion<V> {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        serializer.serialize_u32(V)
+    }
+}
+
+impl<'de, const V: u32> Deserialize<'de> for MetadataVersion<V> {
+    fn deserialize<D: serde::Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        let raw = u32::deserialize(deserializer)?;
+        if raw != V {
+            Err(D::Error::custom(format!(
+                "expected version {V}, found version {raw}"
+            )))
+        } else {
+            Ok(MetadataVersion)
+        }
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "kebab-case")]
+pub struct Metadata {
+    pub release_identifier: String,
+    // TODO: channel should be an enum?
+    pub channel: String,
+    pub ferrocene_version: String,
+    pub rust_version: String,
+    pub release_date: String,
+    pub support_expires_date: String,
+    pub yanked: Option<Yanked>,
+    pub signature: String,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Copy)]
+pub enum MetadataKind {
+    #[serde(rename = "Ferrocene-Release")]
+    FerroceneRelease,
+}
+
+/// Indicates that a version has been yanked, meaning it is considered problematic or potentially unstable.
+/// Yanked versions are not recommended for use in new projects, and developers are strongly advised to avoid them.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct Yanked {
+    /// A machine-readable code representing the reason for yanking.
+    pub code: u16,
+
+    /// A human-readable explanation of why the version was yanked.
+    pub human: String,
```

Comment:
Is it the case that a given code always has the same related `human` string?

Why not name this `reason` or `message`, instead of the unclear `human`?